code/game/objects/items/devices/aicard.dm

```diff
@@ -26,18 +26,16 @@
 	user.visible_message("<span class='suicide'>[user] is trying to upload [user.p_them()]self into [src]! That's not going to work out well!</span>")
 	return BRUTELOSS
 
-/obj/item/aicard/afterattack(atom/target, mob/user, proximity)
-	. = ..()
-	if(!proximity || !target)
-		return
+/obj/item/aicard/pre_attack(atom/target, mob/living/user, params)
 	if(AI) //AI is on the card, implies user wants to upload it.
-		log_combat(user, AI, "carded", src)
 		target.transfer_ai(AI_TRANS_FROM_CARD, user, AI, src)
 	else //No AI on the card, therefore the user wants to download one.
 		target.transfer_ai(AI_TRANS_TO_CARD, user, null, src)
 		if(AI)
-			log_combat(user, AI, "carded", src)
-	update_icon() //Whatever happened, update the card's state (icon, name) to match.
+			log_combat(user, AI, "uploaded", src, "to [target].")
```

Comment:
the wording on this log seems backwards... also the removal of the uploading log, it seems like this message was supposed to go in the first case and should've remained unchanged here